src/gui/editorwidgets/qgsrelationreferencesearchwidgetwrapper.cpp

```diff
@@ -212,6 +212,12 @@ void QgsRelationReferenceSearchWidgetWrapper::initWidget( QWidget *editor )
   mWidget->setAllowAddFeatures( false );
   mWidget->setOpenFormButtonVisible( false );
 
+  const bool fetchLimitActive = config( QStringLiteral( "FetchLimitActive" ), QgsSettings().value( QStringLiteral( "maxEntriesRelationWidget" ), 100, QgsSettings::Gui ).toInt() > 0 ).toBool();
+  if ( fetchLimitActive )
+  {
+    mWidget->setFetchLimit( config( QStringLiteral( "FetchLimitNumber" ), QgsSettings().value( QStringLiteral( "maxEntriesRelationWidget" ), 100, QgsSettings::Gui ) ).toInt() );
+  }
```

Comment:
if fetchLimitActive config is true -> use fetchLimitNumber configuration
otherwise fall back to the QgsSettings value?

Reply:
Not exactly 

If fetchLimitActive is true (because of set in the widgetconfig or QgsSettings value > 0) use fetchLimitNumber (and if no fetchLimitNumber set, fall back to the QgsSettings value).

If fetchLimitActive is false (because of UNset in the widgetconfig or QgsSettings value <=0) use 0 as fetchLimit (means no limit).

See the backwards compatibility note in the PR description.

Reply:
Ok, makes sense 👍